pkg/messagingrp/setup/setup.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Radius Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package setup
+
+import (
+	"time"
+
+	asyncctrl "github.com/radius-project/radius/pkg/armrpc/asyncoperation/controller"
+	"github.com/radius-project/radius/pkg/armrpc/builder"
+	apictrl "github.com/radius-project/radius/pkg/armrpc/frontend/controller"
+	"github.com/radius-project/radius/pkg/messagingrp/datamodel"
+	"github.com/radius-project/radius/pkg/messagingrp/datamodel/converter"
+	"github.com/radius-project/radius/pkg/recipes/controllerconfig"
+
+	msrp_ctrl "github.com/radius-project/radius/pkg/messagingrp/frontend/controller"
+	rmq_ctrl "github.com/radius-project/radius/pkg/messagingrp/frontend/controller/rabbitmqqueues"
+	rmq_proc "github.com/radius-project/radius/pkg/messagingrp/processors/rabbitmqqueues"
+	pr_ctrl "github.com/radius-project/radius/pkg/portableresources/backend/controller"
+	rp_frontend "github.com/radius-project/radius/pkg/rp/frontend"
+)
+
+const (
+	// AsyncOperationRetryAfter is polling interval for async create/update or delete resource operations.
+	AsyncOperationRetryAfter = time.Duration(5) * time.Second
+)
+
+// SetupNamespace builds the namespace for messaging resource provider.
+func SetupNamespace(recipeControllerConfig *controllerconfig.RecipeControllerConfig) *builder.Namespace {
+	ns := builder.NewNamespace("Applications.Messaging")
+
+	_ = ns.AddResource("rabbitMQQueues", &builder.ResourceOption[*datamodel.RabbitMQQueue, datamodel.RabbitMQQueue]{
+		RequestConverter:  converter.RabbitMQQueueDataModelFromVersioned,
+		ResponseConverter: converter.RabbitMQQueueDataModelToVersioned,
+
+		Put: builder.Operation[datamodel.RabbitMQQueue]{
```

Comment:
I see that Put and Patch are identical most of the time and we can just have one definition (like var) for this and pass it into both Put and Patch.

Reply:
We can file an issue for this.